costs/parser.go

```diff
@@ -0,0 +1,114 @@
+//   Copyright 2017 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package costs
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"gopkg.in/olivere/elastic.v5"
+)
+
+func flatSubAggregation(in *interface{}) {
+	oldTab := (*in).([]interface{})
+	newMap := make(map[string]interface{})
+	for _, field := range oldTab {
+		for name, subField := range field.(map[string]interface{}) {
+			newMap[name] = subField
+		}
+	}
+	*in = newMap
+}
+
+func cleanSubAggregation(aggregationName string, aggregationType interface{}, in map[string]interface{}) {
+	key, ok := in["key_as_string"]
+	if !ok {
+		key, ok = in["key"]
+	}
+	skey := fmt.Sprintf("%v", key)
```

Comment:
You only use `skey` in your second `if` block. Why not initialize it there, and transform your `if !ok {} ... if ok {}` by an `if !ok {} else {}` ?